### PR TITLE
[dev] Bug AB#100081: [My Serving Opps][My Ministries>SO] - Previously selected menu option(s) are remining open, when user selects any other menu option

### PR DESCRIPTION
**Pull request by romantech48**
[Bug 100081](https://dev.azure.com/saddlebackchurch/Church%20Management/_workitems/edit/100081): [My Serving Opps][My Ministries>SO] - Previously selected menu option(s) are remining open, when user selects any other menu option

Justifications:
- Unsure why stopPropagation was used, but it stops the event from bubbling, so onClickOutside on the document doesn't fire. 

https://github.com/saddlebackdev/react-cm-ui/assets/103613489/42e36e39-e6e2-4be1-a514-8f0b0f6d6cdb

- Encountered issue with listeners. The issue is that when we return a function it captures `isOpen` value at the moment, so when `isOpen = true` we create our listeners and function to remove them, but as `!true = false`  nothing happens when this function is called

https://github.com/saddlebackdev/react-cm-ui/assets/103613489/ad02770c-1c50-4d85-9a27-28f235041da2

### CODE REVIEW

File: src/inputs/dropdownMenu/dropdownMenu.jsx [hunk 1]
```diff
@@ -245,7 +247,7 @@ function DropdownMenu(props) {
         onDropdownMenuReposition();
 
         return () => {
-            if (!isOpen) {
+            if (isOpen) {
```
Comment (groberts314) [inline src/inputs/dropdownMenu/dropdownMenu.jsx:250]: Can we double-check this?
See https://github.com/saddlebackdev/react-cm-ui/commit/91ba55053be35f638804efca7a97f57c904298fa#diff-cd14cd4dc11800d818e7e59512fcad5a4ec75c5b23cfdaa368b9db636b6165c0L233-R248 / https://github.com/saddlebackdev/react-cm-ui/pull/438.

File: src/inputs/dropdownButton/dropdownButton.jsx [hunk 2]
```diff
@@ -273,10 +270,11 @@ function DropdownButton(props) {
...
 
-    const onMenuToggle = useCallback((event) => {
-        if (!isEmpty(event)) {
-            event.stopPropagation();
-        }
+    const onMenuToggle = useCallback(() => {
+        // TODO: Investigate the reason why stopPropagation used
+        // if (!isEmpty(event)) {
+        //     event.stopPropagation();
+        // }
```
[the remark this answers is not in the document]
Comment (romantech48) [reply src/inputs/dropdownButton/dropdownButton.jsx]: @groberts314 I've added `stopPropagation` option and extended docs with small example

File: src/inputs/dropdownMenu/dropdownMenu.jsx (hunk 1 above)
Comment (romantech48) [reply src/inputs/dropdownMenu/dropdownMenu.jsx:250]: @groberts314 I've double checked, with current `(!isOpen)`, listeners are not removed on dev, stage, stage2. If needed, I can decompose the explanation of why it's not working from the description into an ordered list

**Review by War777 — APPROVED** [review]
Looks good to me, just make sure to do an integrated testing across a few places in HC please to make sure we're not regressing anything.
Thanks @romantech48 .

**Review by groberts314 — APPROVED** [review]

**Pull request merged**